### DISCUSSION TITLE
CropOrPad Function Problem

### DISCUSSION THREAD

**Discussion by oguzcanbekar** [TorchIO-project/torchio#606] [Q&A]
Hi there @fepegar I have just met with TorchIO nowadays and I am really thankful for your great work. Basically I would like to explain my problem and ask my question. I have just tried CropOrPad function on 2 different CT images but when the function needs to crop, everything was worked very well but when I am trying to padding(I mean my target shape size lesser than image shape) the image and mask result couldn`t match.I also tried some another padding modes but still the result didn`t change. I will share code and results, hopefully we can figure it out easily. Thank you very much again.
![code_part](https://user-images.githubusercontent.com/52401130/125555695-2c3bb272-5108-44a9-9798-bee71f2e56d2.png)
![crop image](https://user-images.githubusercontent.com/52401130/125555699-016dfc8c-6583-4224-8787-052347fe4525.png)
![pad_image_result](https://user-images.githubusercontent.com/52401130/125555701-b60b1b3d-39f9-4716-812e-e8f8adcc979d.png)

**Comment by romainVala** [comment]
hello
that is weird, there is no reason that CropOrPad transform will shift the mask differently thant the subject image ...
Are you sure both image and mask match before pading ?

**Comment by fepegar** [comment]
Hi, @oguzcanbekar. Maybe your images are not in the same physical space. Could you please share them here?
